remotes/docker/resolver.go

```diff
@@ -130,8 +136,23 @@ type dockerResolver struct {
 	header        http.Header
 	resolveHeader http.Header
 	tracker       StatusTracker
+	trace         bool
 }
 
+var (
+	tracer = &httptrace.ClientTrace{
+		DNSStart: func(dnsInfo httptrace.DNSStartInfo) {
+			log.L.Debugf("DNSStart: %s, lookup: %s", time.Now().UTC().String(), dnsInfo.Host)
+		},
+		DNSDone: func(dnsInfo httptrace.DNSDoneInfo) {
+			log.L.Debugf("DNSEnd:   %s, result: %s, err? %v, coalesced: %v", time.Now().UTC().String(), dnsInfo.Addrs[0].String(), dnsInfo.Err, dnsInfo.Coalesced)
+		},
+		GotConn: func(connInfo httptrace.GotConnInfo) {
+			log.L.Debugf("GotConn:  %s, reused? %v", time.Now().UTC().String(), connInfo.Reused)
+		},
+	}
+)
+
```

Comment:
I agree with @cpuguy83. The default tracer is fine for `ctr` but assuming all containerd clients need these three events looks a bit odd.